### PR TITLE
[REVIEW] allow building nvcomp with ptds [skip ci]

### PR DESCRIPTION
**Pull request by rongou**
This should build `nvcomp` with per-thread default stream when specified from maven.

@jlowe @abellina

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: java/src/main/native/cmake/Modules/ConfigureNvcomp.cmake [hunk 1]
```diff
@@ -28,6 +28,12 @@ elseif(CMAKE_CXX11_ABI)
     list(APPEND NVCOMP_CMAKE_ARGS " -DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=1")
 endif(NOT CMAKE_CXX11_ABI)
 
+if(PER_THREAD_DEFAULT_STREAM)
+    message(STATUS "NVCOMP: Enabling per-thread default stream")
+    list(APPEND NVCOMP_CMAKE_ARGS " -DCMAKE_C_FLAGS=-DCUDA_API_PER_THREAD_DEFAULT_STREAM")
+    list(APPEND NVCOMP_CMAKE_ARGS " -DCMAKE_CXX_FLAGS=-DCUDA_API_PER_THREAD_DEFAULT_STREAM")
```
Comment (jlowe) [inline java/src/main/native/cmake/Modules/ConfigureNvcomp.cmake:34]: Will this approach work if nvcomp's CMakeLists.txt sets either of these variables?  It does so conditionally as of the 1.1 release.  I'm worried about this working today but then silently not working if nvcomp updates their CMakeLists.txt in the future to unconditionally set these variables and clobber any setting we're trying to specify on the command-line.

**Comment by abellina** [comment]
@rongou I am going to test your change with my setup, since I saw this issue originally.

**Comment by rongou** [comment]
Superseded by #6690

**Pull request closed without merging**